src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/XGBoostEvidenceFilter.java

```diff
@@ -0,0 +1,609 @@
+package org.broadinstitute.hellbender.tools.spark.sv.evidence;
+
+import biz.k11i.xgboost.Predictor;
+import biz.k11i.xgboost.learner.ObjFunction;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Lists;
+import htsjdk.samtools.CigarElement;
+import htsjdk.samtools.CigarOperator;
+import htsjdk.samtools.TextCigarCodec;
+import htsjdk.tribble.AbstractFeatureReader;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.spark.sv.StructuralVariationDiscoveryArgumentCollection;
+import org.broadinstitute.hellbender.tools.spark.sv.utils.*;
+import org.broadinstitute.hellbender.tools.spark.utils.IntHistogram;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+import scala.Tuple2;
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * A class that acts as a filter for BreakpointEvidence.
+ * Features are calculated according to evidence type, overlap information, mapping quality, etc.
+ * A trained classifier scores the probability the evidence overlaps a breakpoint interval, and passes evidence above
+ * the specified threshold.
+ */
+public final class XGBoostEvidenceFilter implements Iterator<BreakpointEvidence> {
+    static private final boolean useFastMathExp = true;
+    private static final List<String> defaultEvidenceTypeOrder = Arrays.asList(
```

Comment:
The order matters because the classifier can only look at doubles, and the encoding used by the training routines must be identical to that used by the evaluation. I've come to the conclusion that the override capability is confusing overkill so I've removed it. Just providing a list to specify order seemed like the simplest way to provide the order that had an initializer in java. I'm happy to change it to something simpler/better in response to follow-on comments or a chat.